File: data-processing/src/review.js
/** 
 * Generic format for a movie review
*/
class review {
    #movieID;
    #rawText;
    #sentiments;

    constructor(movieID, text){
        this.#movieID = movieID;
        this.#rawText = text;
        this.#analyze();
    }

    /**
     * Performs sentiment analysis on the raw text and stores it in sentiments
     */
    #analyze(){
        //TO-DO 
    }

    get movieID() {
        return this.#movieID;
    }

    get rawText() {
        return this.#rawText;
    }

    get sentiments() {
        return this.#sentiments;
    }
}

File: data-processing/src/review.py
from analyzer import Analyzer

# Required information for a review
class Review:

    # Constructor. Performs sentiment analysis on the raw text
    def __init__(self, score, raw_text) -> None:
        self.score = score
        self.raw_text = raw_text

        #sentiment analysis
        a = Analyzer()
        self.sentiments = a.analyze(raw_text)